src/test/java/edu/harvard/iq/dataverse/api/UtilIT.java

```diff
@@ -2577,9 +2582,24 @@ static boolean sleepForReexport(String idOrPersistentId, String apiToken, int du
         return i <= repeats;
 
     }
-    
-    
-    
+
+    // Modeled after sleepForLock but the dataset isn't locked.
+    // We have to sleep or we can't perform the next operation.
+    static Boolean sleepForDeadlock(int duration) {
+        int i = 0;
+        do {
+            try {
+                Thread.sleep(1000);
+                i++;
+                if (i > duration) {
+                    break;
+                }
+            } catch (InterruptedException ex) {
+                Logger.getLogger(UtilIT.class.getName()).log(Level.SEVERE, null, ex);
+            }
+        } while (true);
+        return i <= duration;
+    }
```

Comment:
This sleepForDeadlock is used in DataversesIT. Details:

commit 7b4f827af1a885a2560ba717b52d7d34fe03ab2c
Author: Philip Durbin <philip_durbin@harvard.edu>
Date:   Fri Jun 3 12:47:28 2022 -0400

    stop using `@Context HttpServletRequest`, add sleep #8305
    
    In EditDDI I was seeing a NullPointerException on line 136
    (createNewDraftVersion). Switch from `@Context HttpServletRequest`
    to our ordinary way of API auth fixed it.
    
    In DataversesIT I was seeing an OptimisticLockException so I
    added a sleep.